### PR TITLE
Adds support for angles > 180 and snapping to lower and upper limits of grid angle range

### PR DESCRIPTION
**Pull request by hamzahasann**
https://github.com/oriedita/oriedita/assets/64196922/8823072d-715e-42eb-8cea-a86b9dd41cda

### CODE REVIEW

File: origami/src/main/java/origami/Epsilon.java [hunk 1]
```diff
@@ -30,13 +30,12 @@ public class Epsilon {
...
     public static final double UNKNOWN_1EN6 = factor * 1E-6;
     public static final double UNKNOWN_1EN7 = factor * 1E-7;
-
     // These are the constants with a known purpose.
 
     public static final double PARALLEL = factor * 0.5;
     public static final double FLAT = factor * 1E-4;
     public static final double QUAD_TREE_ITEM = factor * 0.5;
-
+    public static final double GRID_ANGLE_THRESHOLD = 5;
```
Comment (undertrox) [inline origami/src/main/java/origami/Epsilon.java:38]: I think 5 might be a bit too much, maybe 1 or even 0.5 would be better.

**Review by undertrox — CHANGES_REQUESTED** [review]
I like the changes a lot, just that one little change and it's ready to merge

File: origami/src/main/java/origami/Epsilon.java (hunk 1 above)
Comment (hamzahasann) [reply origami/src/main/java/origami/Epsilon.java]: Updated the threshold to 1.

**Review by undertrox — APPROVED** [review]

File: origami/src/main/java/origami/Epsilon.java (hunk 1 above)
Comment (TabLand) [reply origami/src/main/java/origami/Epsilon.java]: Smaller angles also cause performance issues on slow machines; e.g mine hangs at 1 degree; and the lines look almost much parallel at 10x zoom

**Pull request merged**